Otus-DevOps-2018-02/ifqthenp_infra
# Homework 9. Introduction to Ansible

# Completed Homework № 9

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

:large_blue_diamond: Installed Ansible control machine to execute Ansible tasks from
:large_blue_diamond: Created inventory files in various formats: `ini`, `yml`, `json`
:large_blue_diamond: Created `ansible.cfg` config file with default values for the project
:large_blue_diamond: Created simple ansible playbook to test ansible behaviour for repeating tasks
:large_orange_diamond: Created simple ansible playbook, described in `clone.yml`, to test ansible behaviour for repeating tasks. The tests have shown that multiple identical commands sent with ansible to the server have the same effect as single command sent to server only once, i.e. ansible module `command` is idempotent.

## How to get project running

- [Install Ansible](http://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html)
- Deploy infrastructure described in `stage` folder by using `terraform apply` command
- Enjoy

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run `ansible all -m ping` command to see if ping is successful

## PR checklisk
